src/Glpi/Form/Destination/CommonITILField/AssociatedItemsFieldStrategy.php

```diff
@@ -132,62 +145,82 @@ private function getAssociatedItemsForSpecificAnswer(
             return null;
         }
 
-        $value = $answer->getRawAnswer();
-        if (!$this->isValidAnswer($value)) {
+        $values = $answer->getRawAnswer();
+        if (count($values) == count($values, COUNT_RECURSIVE)) {
+            $values = [$values];
+        }
+
+        $values = array_filter($values, fn($value) => $this->isValidAnswer($value));
+        if (empty($values)) {
             return null;
         }
 
-        return $value;
+        return $values;
     }
 
     private function getAssociatedItemsForLastValidAnswer(
         AnswersSet $answers_set,
     ): ?array {
-        $valid_answers = array_filter(
-            $answers_set->getAnswersByType(
-                QuestionTypeItem::class
-            ),
-            fn($answer) => in_array(
-                $answer->getRawAnswer()['itemtype'],
-                CommonITILObject::getAllTypesForHelpdesk()
-            )
-        );
+        $valid_answers = $this->getValidAnswers($answers_set);
 
         if (count($valid_answers) == 0) {
             return null;
         }
 
         $answer = end($valid_answers);
-        $value = $answer->getRawAnswer();
-        if (!$this->isValidAnswer($value)) {
+        $values = $answer->getRawAnswer();
+        if (count($values) == count($values, COUNT_RECURSIVE)) {
+            $values = [$values];
+        }
+
+        if (empty($values)) {
             return null;
         }
 
-        return [$value];
+        return $values;
     }
 
     private function getAssociatedItemsFromAllValidAnswers(
         AnswersSet $answers_set,
     ): ?array {
-        $valid_answers = array_filter(
-            $answers_set->getAnswersByType(
-                QuestionTypeItem::class
-            ),
-            fn($answer) => in_array(
-                $answer->getRawAnswer()['itemtype'],
-                CommonITILObject::getAllTypesForHelpdesk()
-            )
-        );
+        $valid_answers = $this->getValidAnswers($answers_set);
 
         if (count($valid_answers) == 0) {
             return null;
         }
 
-        $associted_items = array_map(
-            fn($answer) => $answer->getRawAnswer(),
+        return array_merge(...array_map(
+            function ($answer) {
+                $raw_answer = $answer->getRawAnswer();
+                if (count($raw_answer) == count($raw_answer, COUNT_RECURSIVE)) {
+                    $raw_answer = [$raw_answer];
+                }
+
+                return $raw_answer;
+            },
             $valid_answers
-        );
+        ));
```

Comment:
IMHO, you may have the same item multiple times if the same item is present in multiple answers.